### PR TITLE
Increasing code coverage

### PR DESCRIPTION
**Pull request by pk-druva**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Druva/Integrations/Druva/Druva_test.py [hunk 1]
```diff
@@ -1,3 +1,75 @@
...
+            'Authentication': 'Bearer some_api_key'
+        }
+    )
+    
+    with open("test_data/test-data.json") as file:
+        data=json.load(file)
+        commands=["test-module","druva-list-quarantine-ranges","druva-find-device","druva-find-user","druva-find-userDevice","druva-find-sharePointSites","druva-find-sharedDrives",
+                  "druva-quarantine-resource","druva-delete-quarantine-range","druva-view-quarantine-range","druva-update-quarantine-range","druva-list-quarantine-snapshots","druva-delete-quarantined-snapshot",
+                  "druva-endpoint-check-restore-status","druva-endpoint-search-file-hash"]
+        for cur_command in commands:
```
Comment (dantavori) [inline Packs/Druva/Integrations/Druva/Druva_test.py:44]: (this note is not related to the suggestion above)
your current approach is totally ok, but FYI: you can avoid defining the `commands` list and iterating it by doing something like:
```suggestion
        for command_name, command_data in data.items():
            mocker.patch.object(demisto, "command", return_value=command_name)
            if get_request_url := command_data["get-request-url"]:
                mock_response = command_data["get-response-mock-data"]
                requests_mock.get(get_request_url, status_code=200, json=mock_response)
            if post_request_url := command_data["post-request-url"]:
                mock_response = command_data["post-response-mock-data"]
                requests_mock.post(post_request_url, status_code=200, json=mock_response)
            if put_request_url := command_data["put-request-url"]:
                mock_response = command_data["put-response-mock-data"]
                requests_mock.put(put_request_url, status_code=200, json=mock_response)
            if delete_request_url := command_data["delete-request-url"]:
                mock_response = command_data["delete-response-mock-data"]
                requests_mock.delete(delete_request_url, status_code=200, json=mock_response)
            main()
```
of course, this requires some additional changes in the test-data.json file. As I mentioned above, this change is not required, only my recommendation :)

File: Packs/Druva/Integrations/Druva/Druva_test.py [hunk 2]
```diff
@@ -1,3 +1,75 @@
...
+                Execute_command(cur_command,client)
+            elif req_type=="delete":
+                mock_response=data[cur_command]["mock-responce"]
+                requests_mock.delete(data[cur_command]["url-suffix"],status_code=200,json=mock_response)
+                Execute_command(cur_command,client)
+            elif req_type=="post":
+                mock_response=data[cur_command]["mock-responce"]
+                requests_mock.delete(data[cur_command]["url-suffix"],status_code=200,json=mock_response)
+                Execute_command(cur_command,client)
+
```
Comment (dantavori) [inline Packs/Druva/Integrations/Druva/Druva_test.py:70]: it is recommended to make some assertions at the end of unit tests.
for this, you can do something like:
```suggestion
        outputs = demisto.results.call_args[0][0]['Contents']
        assert something_about_outputs
```
if you want to do this, you should run at the beginning of the method:
```
mocker.patch.object(demisto, 'results')
```

**Review by dantavori — CHANGES_REQUESTED** [review]
hey there, it is a great idea to add tests to your code :)
see my notes below. I hope I am not too tough with the review, I assumed the goal is to make valuable tests so I added some suggestions for improvements. If something in my notes is not clear, feel free to reach me.

Note that you have some lint (flake8) errors - https://app.circleci.com/pipelines/github/demisto/content/253805/workflows/407760e9-6bb0-40a8-b93d-9d58794f7424/jobs/697310
To fix them, you can `autopep8 -i <path_to_script>` on the modified scripts.

File: Packs/Druva/Integrations/Druva/Druva_test.py (hunk 2 above)
Comment (pk-druva) [reply Packs/Druva/Integrations/Druva/Druva_test.py:70]: Hi,dantavori
What's problem to have separate Execute_command.

Comment (dantavori) [reply Packs/Druva/Integrations/Druva/Druva_test.py:70]: I explain in the first comment: https://github.com/demisto/content/pull/26060#discussion_r1175542485
It's not a problem, but if you insist on it, you must bump a version for the pack and add a release note.

**Comment by kgal-pan** [comment]
Hi @pk-druva - Any updates on this?

**Comment by kgal-pan** [comment]
Closing as there's been no reply since May 2nd.

**Comment by pk-druva** [comment]
Hi, @kgal-pan sorry i was on leave for month for some personal reason just saw your comments today 
But i did addressed all the comments requested by @dantavori  before leaving 
can you please re-open this request and contact me for any further assistance?

File: Packs/Druva/Integrations/Druva/Druva_test.py (hunk 2 above)
Comment (pk-druva) [reply Packs/Druva/Integrations/Druva/Druva_test.py]: Bumped the version by note as Internal code improvements.

**Comment by pk-druva** [comment]
Hi @kgal-pan and @dantavori 
Is there any review comments that i have't addressed yet?

**Comment by kgal-pan** [comment]
> Hi @kgal-pan and @dantavori Is there any review comments that i have't addressed yet?

You have a merge conflict, please resolve it. There are also 2 validation errors that need to be addressed, likely related to the merge conlict. https://app.circleci.com/pipelines/github/demisto/content/255564/workflows/04f5b383-da32-4134-8dfe-c004447ffe12/jobs/698148

**Comment by pk-druva** [comment]
> > Hi @kgal-pan and @dantavori Is there any review comments that i have't addressed yet?
> 
> You have a merge conflict, please resolve it. There are also 2 validation errors that need to be addressed, likely related to the merge conflict. https://app.circleci.com/pipelines/github/demisto/content/255564/workflows/04f5b383-da32-4134-8dfe-c004447ffe12/jobs/698148


I think after creation of MR release notes are updated so what should be done here should i upgrade release notes once again?

**Comment by kgal-pan** [comment]
@pk-druva - Looks like you generated the release notes manually. We have 2 options:

1) Remove the `Packs/Druva/ReleaseNotes/1_1_7.md` and run:

```bash
demisto-sdk update-release-notes -i Packs/Druva -u revision
```

2) Manually change `Packs/Druva/pack_metadata.json` `version` field to `1.1.7`.

**Comment by dorschw** [comment]
Closing due to inactivity, feel free to reopen
